scripts/semver-check.sh

```diff
@@ -37,23 +38,19 @@ if [[ ! -z ${TAGGED_RELEASE_VERSION} ]] && [[ ! ${TAGGED_RELEASE_VERSION} =~ "rc
         exit 1
     fi
 
-    if [[ ${API_COMPAT_LEVEL} == "major" ]] && [[ ${newmajor} != $((oldmajor+1)) ]]; then
-        echo "Major API breaking change. Major version number must be incremented."
+    if [[ ${newMajor} == $((oldMajor + 1)) ]]; then
+        echo "Major API breaking change."
+    elif [[ ${newMajor} > $((oldMajor + 1)) ]] || [[ ${newMajor} < $((oldMajor)) ]] || [[ ${API_COMPAT_LEVEL} == "major" && ${newMajor} != $((oldMajor + 1)) ]]; then
+        echo "Major version number changed incorrectly."
         exit 1
-    elif [[ ${API_COMPAT_LEVEL} == "minor" ]] && [[ ${newMinor} != $((oldMinor+1)) ]]; then
-        echo "Minor API breaking change. Minor version number must be incremented."
-        exit 1
-    elif [[ ${API_COMPAT_LEVEL} == "patch" ]] && [[ ${newPatch} != $((oldPatch+1)) ]]; then
-        echo "Patch version number must be incremented."
-        exit 1
-    else
-        echo "Release tagged with correct version."
-    fi
-else
-    echo "Checked API breaks for branch or release candidate."
-    if [[ ${API_COMPAT_LEVEL} == "major" ]] || [[ ${API_COMPAT_LEVEL} == "minor" ]];then
-        echo "Current version has (${API_COMPAT_LEVEL}) API changes comparing to ${LAST_VERSION}"
-    else
-        echo "Current version is a (${API_COMPAT_LEVEL}) release for the ${LAST_VERSION}"
+    elif [[ ${API_COMPAT_LEVEL} != "major" ]]; then
+        if [[ ${newMinor} < $((oldMinor)) ]]; then
```

Comment:
```suggestion
        if [[ ${newMinor} <= $((oldMinor)) ]]; then
```

Reply:
In line 50 there is a condition check for equals